yang/frr-bfdd.yang

```diff
@@ -233,6 +233,75 @@ module frr-bfdd {
     }
   }
 
+  typedef bfd-source-address {
+    type union {
+      type inet:ip-address;
+      type enumeration {
+        enum "auto";
```

Comment:
I don't think we need to complicate things with this. Just make `leaf source` on line 256 not mandatory and apply automatic rules when it's not set.